test/e2e/storage/flexvolume.go

```diff
@@ -39,7 +39,7 @@ const (
 	// On gci, root is read-only and controller-manager containerized. Assume
 	// controller-manager has started with --flex-volume-plugin-dir equal to this
 	// (see cluster/gce/config-test.sh)
-	gciVolumePluginDir = "/etc/srv/kubernetes/kubelet-plugins/volume/exec"
```

Comment:
Is this going to break skew tests (where we run new tests against old releases and old tests against new releases)?

Reply:
Good point, yeah the new path is not usable in older versions. Will add a check for master and node versions.

Reply:
AFAIU master upgrades leave most of the environment variables unchanged, even if the variables in the new version of GCE startup scripts have different values. If this is true, then I believe it's not necessary to gate the path by master/node version?

Reply:
I'm not entirely clear on what this test is doing (or testing for) but if it's connecting to machines running at different versions then the location of the volume plugin directory will be different on different versions. 

For both masters and nodes, when we upgrade / downgrade we use the environment variables for the desired version (e.g. a 1.9 gci vm would use `/etc/srv/kubernetes/kubelet-plugins/volume/exec` and a 1.10 gci vm would use `/home/kubernetes/flexvolume`. 

There are directions for how we used to do manual upgrade / downgrade testing in https://docs.google.com/document/d/1DtQFhxmKSZJJ_yv8ttweqotburHHZWxaCYnFbjLDA5g/edit which you could use to run this specific test against skewed clusters. If they pass, then this PR is ok. If not, we'll need to update the test code to pick the path based on the k8s version.

Reply:
ACK, added logic for checking cluster version before setting path. Haven't verified this code yet as kubetest has a bug atm (https://github.com/kubernetes/test-infra/issues/6992); will verify once it's fixed.

Reply:
Verified with 1.10 gci master and 1.9 gci nodes